sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -393,6 +393,9 @@ trait OrcCommonFunctions extends OrcCodecWritingHelper {
 
   /**
    * Cast columns with precision that can be stored in an int to DECIMAL32, to save space.
+   * Besides the plugin makes the assumption that if the precision is small enough to fit
+   * in a DECIMAL32, then CUDF has it stored as a DECIMAL32. Getting this wrong may lead
```

Comment:
nit: This is not 100% accurate.  In the plugin we enforce that if the precision is small enough to fit in a DECIMAL32 then it will be stored in a DECIMAL32. We don't assume that CUDF has stored it that way, as we tell CUDF what to do and what types to use.  It just happens that in some cases, like ORC and Parquet we don't have the ability to tell them what to do, so we can end up needing to fix it after the fact.